linkerd/drain/src/lib.rs

```diff
@@ -96,6 +96,15 @@ impl Watch {
         }
     }
 
+    pub async fn and_drop<A, F>(self, future: A, on_drain: F) -> A::Output
+    where
+        A: Future + Unpin,
+        F: FnOnce() -> A::Output,
+    {
+        self.watch(future, |_| {}).await;
+        on_drain()
+    }
```

Comment:
Eh, but this is weird because it potentially drops the return value of `future`... I can fix it..